.github/workflows/main.yml

```diff
@@ -1,6 +1,33 @@
 name: main
 
-on: [push, pull_request]
+on:
+  workflow_call:
+    inputs:
+      build-artifact-name:
+        required: true
+        type: string
+      build-artifact-path:
+        required: true
+        type: string
+      self-build:
+        required: false
+        type: number
+      pyodide-versions:
+        required: false
+        type: string
+        default: '*'
+      runners:
+        required: false
+        type: string
+        default: '*'
+      browsers:
+        required: false
+        type: string
+        default: '*'
```

Comment:
I think by default it might make sense to run only run a subset of these cases. At least for sure people shouldn't run their tests both in selenium and playwright. And also I'm not sure running the full matrix of pyodide-versions x browsers is really worth it.